server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -289,13 +289,13 @@ PublishWithJoinResponse handleIncomingRemotePublishRequest(RemotePublishRequest
                 transportService.getLocalNode().getId()
             );
             compatibleClusterStateDiffReceivedCount.incrementAndGet();
-            final PublishWithJoinResponse response = acceptState(clusterState);
+            final PublishWithJoinResponse response = acceptState(clusterState, manifest);
             lastSeenClusterState.compareAndSet(lastSeen, clusterState);
             return response;
         }
     }
 
-    private PublishWithJoinResponse acceptState(ClusterState incomingState) {
+    private PublishWithJoinResponse acceptState(ClusterState incomingState, ClusterMetadataManifest manifest) {
```

Comment:
Why clusterState is needed?

Reply:
This is common method to apply the state for Remote enabled and disabled. ClusterState was anyway required to be passed here.